apps/nextjs-website/src/components/Header.tsx

```diff
@@ -106,21 +116,47 @@ const makeMegaHeaderProps = ({
     primary: link.label,
     secondary: link.sublinkGroups.map((sublinkGroup) => ({
       title: sublinkGroup.title,
-      items: sublinkGroup.sublinks.map(makeSublink),
+      items: sublinkGroup.sublinks.map((sublink) =>
+        makeSublink(sublink, locale),
+      ),
     })),
+    ...(link.ctaButton && {
+      ctaButton: {
+        ...link.ctaButton,
+        ...(link.ctaButton.icon && { startIcon: Icon(link.ctaButton.icon) }),
+      },
+    }),
   })),
 });
 
-const Header = (props: HeaderData['data']['attributes']['header'][0]) => {
-  const pathname = usePathname();
+const Header = (
+  props: HeaderData['data']['attributes']['header'][0] & {
+    locales: Array<Locale>;
```

Comment:
```suggestion
    locale: Locale;
```
I see you took this implementation from the Footer, but that component has to implement a language switch, hence the need to pass all locales (even then, the implementation of activeLocale seems unnecessary and we should be able to just pass the locale).

Header does not need to have access to all selected locales for building.